staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1567,41 +1438,15 @@ func TestNeedsUpdate(t *testing.T) {
 		{
 			testName: "If appProtocol is the same",
 			updateFn: func() {
-				oldSvc = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "tcp-service",
-						Namespace: "default",
-					},
-					Spec: v1.ServiceSpec{
-						Ports: []v1.ServicePort{{
-							Port:       80,
-							Protocol:   v1.ProtocolTCP,
-							TargetPort: intstr.Parse("22"),
-						}},
-						Type: v1.ServiceTypeLoadBalancer,
-					},
-				}
+				oldSvc = newService("tcp-service", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22))
 				newSvc = oldSvc.DeepCopy()
 			},
 			expectedNeedsUpdate: false,
 		},
 		{
 			testName: "If appProtocol is set when previously unset",
 			updateFn: func() {
-				oldSvc = &v1.Service{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "tcp-service",
-						Namespace: "default",
-					},
-					Spec: v1.ServiceSpec{
-						Ports: []v1.ServicePort{{
-							Port:       80,
-							Protocol:   v1.ProtocolTCP,
-							TargetPort: intstr.Parse("22"),
-						}},
-						Type: v1.ServiceTypeLoadBalancer,
-					},
-				}
+				oldSvc = newService("tcp-service", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22))
 				newSvc = oldSvc.DeepCopy()
```

Comment:
This pattern of DeepCopy and change could also be cleaned up (different commits or PRs).  You could do:

```
newSvc = copyService(oldSvc, tweakAppProtocol("http"))
```

or

```
newSvc = tweakAppProtocol("http")(oldSvd.DeepCopy())
```

but I like the first one better.